### ISSUE TITLE
Slightly different image width between Mac and Linux

### ISSUE THREAD

**Issue by mikecharles** [matplotlib/basemap#313]
When I create a Basemap and save it with `matplotlib.pyplot.savefig()` with the `box_inches='tight'` parameter, I get a slightly different image size on Linux vs. Mac.

### Machines

Machine1: RedHat Enterprise Linux 6.8
Machine2: Mac OS 10.11.6

### Python

I setup an identical Anaconda environment on each:

    $ conda create --no-default-packages -n basemap-test python=3.5 basemap nomkl

Resulting environment:

```
conda list
# packages in environment at /Applications/miniconda/envs/basemap-test:
#
appnope                   0.1.0                     <pip>
basemap                   1.0.7               np111py35_0    defaults
cycler                    0.10.0                   py35_0    defaults
decorator                 4.0.10                    <pip>
freetype                  2.5.5                         1    defaults
geos                      3.4.2                         0    conda-forge
ipdb                      0.10.1                    <pip>
ipython                   5.0.0                     <pip>
ipython-genutils          0.1.0                     <pip>
libpng                    1.6.22                        0    defaults
matplotlib                1.5.1               np111py35_0    defaults
nomkl                     1.0                           0    defaults
numpy                     1.11.1             py35_nomkl_0  [nomkl]  defaults
openssl                   1.0.2h                        1    defaults
pexpect                   4.2.0                     <pip>
pickleshare               0.7.3                     <pip>
pip                       8.1.2                    py35_0    defaults
prompt-toolkit            1.0.3                     <pip>
ptyprocess                0.5.1                     <pip>
pygments                  2.1.3                     <pip>
pyparsing                 2.1.4                    py35_0    defaults
pyqt                      4.11.4                   py35_4    defaults
python                    3.5.2                         0    defaults
python-dateutil           2.5.3                    py35_0    defaults
pytz                      2016.6.1                 py35_0    defaults
qt                        4.8.7                         4    defaults
readline                  6.2                           2    defaults
setuptools                23.0.0                   py35_0    defaults
simplegeneric             0.8.1                     <pip>
sip                       4.18                     py35_0    defaults
six                       1.10.0                   py35_0    defaults
sqlite                    3.13.0                        0    defaults
tk                        8.5.18                        0    defaults
traitlets                 4.2.2                     <pip>
wcwidth                   0.1.7                     <pip>
wheel                     0.29.0                   py35_0    defaults
xz                        5.2.2                         0    defaults
zlib                      1.2.8                         3    defaults
```

## Script

Here's the script I ran on both machines:

```python
from mpl_toolkits.basemap import Basemap
import matplotlib.pyplot as plt


fig, ax = plt.subplots()

basemap = Basemap(width=8000000, height=7500000, lat_0=48., lon_0=260.,
                  projection='laea', ax=ax, resolution='l')

basemap.drawcountries(linewidth=0.5)
basemap.drawcoastlines(0.5)

plt.savefig('test.png', dpi=100, bbox_inches='tight')
```

And here are the resulting images:

Linux:
![linux](https://cloud.githubusercontent.com/assets/265620/17296628/384be496-57d0-11e6-8bc1-238884516097.png)

Mac:
![mac](https://cloud.githubusercontent.com/assets/265620/17296634/3bc09b1c-57d0-11e6-8867-84c25e7d0918.png)

Though they look identical, they are 1 pixel different in width:

    $ identify linux.png mac.png
    linux.png PNG 531x500 531x500+0+0 8-bit sRGB 85.5KB 0.000u 0:00.000
    mac.png PNG 532x500 532x500+0+0 8-bit sRGB 85.5KB 0.000u 0:00.000

I don't know if this is a `basemap` issue or a `matplotlib` issue.

**Comment by WeatherGod** [comment]
Just a shot in the dark, but could you try the latest beta of matplotlib
v2.0? You can get it via pip using the --pre option. The macosx backend was
completely reworked and it fixed many inconsistencies. I am not exactly
sure if this would be one of them since most of them had to do with the
display rather than the saving, but it might be worth a try.

**Comment by mikecharles** [comment]
Well, I'm having some trouble with v2.0.

### Mac

On the Mac, I installed v2.0.0b3 like this:

    pip install -U --pre matplotlib

Then when I run the code above I get this error:

```
Traceback (most recent call last):
  File "test.py", line 2, in <module>
    import matplotlib.pyplot as plt
  File "/Applications/miniconda/envs/basemap-test/lib/python3.5/site-packages/matplotlib/pyplot.py", line 114, in <module>
    _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
  File "/Applications/miniconda/envs/basemap-test/lib/python3.5/site-packages/matplotlib/backends/__init__.py", line 32, in pylab_setup
    globals(),locals(),[backend_name],0)
  File "/Applications/miniconda/envs/basemap-test/lib/python3.5/site-packages/matplotlib/backends/backend_macosx.py", line 19, in <module>
    from matplotlib.backends import _macosx
RuntimeError: Python is not installed as a framework. The Mac OS X backend will not be able to function correctly if Python is not installed as a framework. See the Python documentation for more information on installing Python as a framework on Mac OS X. Please either reinstall Python as a framework, or try one of the other backends. If you are Working with Matplotlib in a virtual enviroment see 'Working with Matplotlib in Virtual environments' in the Matplotlib FAQ
```

I also tried installing v2.0 first (instead of updating), but when I go to install basemap with Anaconda it wants to downgrade Matplotlib. I'll look at the FAQ soon...

### Linux

On Linux, after I update Matplotlib the script runs with no complaints, but the file comes out with a completely different size:

Matplotlib v1.5.1
![linux-v1 5 1](https://cloud.githubusercontent.com/assets/265620/17330772/296eef24-5896-11e6-9328-78904701635f.png)

Matplotlib v2.0.0b3
![linux-v2 0 0](https://cloud.githubusercontent.com/assets/265620/17330790/3665027c-5896-11e6-845e-74eb1d3faf42.png)

**Comment by jenshnielsen** [comment]
On Anaconda you should be able to install python.app and and use pythonw instead of python to get a framework build. This is also an issue with matplotlib 1.5.1 but Anaconda patched out the check. The non framework build produces figures that can't disappear behind other windows and can't probably be moved to the foreground

**Comment by WeatherGod** [comment]
The new filesize is a result of the change in the default DPI and is to be
expected. What will be important is comparing both the mac and the linux
produced files.

**Comment by jenshnielsen** [comment]
If you are just interested in a png you can also disable the gui backend. Add the following to the top to select a non gui backend.

```
import matplotlib
matplotlib.use('Agg')
```

**Comment by mikecharles** [comment]
Ok I got it working on the Mac with this:

    $ export MPLBACKEND=Agg

Now Linux and Mac both produce identical plots, so I guess it'll be fixed in v2.0.

Thanks everybody
